marketplace/google-analytics-app/src/AppConfig.tsx

```diff
@@ -0,0 +1,342 @@
+import * as React from 'react';
+import sortBy from 'lodash/sortBy';
+import {
+  Typography,
+  Heading,
+  Paragraph,
+  TextField,
+  Option,
+  TextLink,
+  Button,
+  Select,
+  FormLabel,
+  TextInput
+} from '@contentful/forma-36-react-components';
+import styles from './styles';
+import { AppConfigParams, AppConfigState } from './typings';
+import { getAndUpdateSavedParams } from './utils';
+import {
+  ContentType,
+  EditorInterface,
+  ContentTypeField,
+  CollectionResponse
+} from 'contentful-ui-extensions-sdk';
+
+export default class AppConfig extends React.Component<AppConfigParams, AppConfigState> {
+  state: AppConfigState = {
+    allContentTypes: {},
+    contentTypes: {},
+    clientId: '',
+    viewId: ''
+  };
+
+  async componentDidMount() {
+    const { sdk } = this.props;
+
+    const [{ items: spaceContentTypes }, savedParams] = await Promise.all([
+      sdk.space.getContentTypes() as Promise<
+        CollectionResponse<ContentType & { fields: ContentTypeField[] }>
+      >,
+      getAndUpdateSavedParams(sdk)
+    ]);
+
+    const { contentTypes } = savedParams;
+
+    // add an incomplete contentType entry if there are none saved
+    if (!Object.keys(contentTypes).length) {
+      contentTypes[''] = { slugField: '', urlPrefix: '' };
+    }
+
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState(
+      {
+        // sort contentTypes by display name
+        allContentTypes: sortBy(spaceContentTypes, 'name').reduce(
+          (acc: AppConfigState['allContentTypes'], contentType) => {
```

Comment:
I'd introduce a separate type instead of using `AppConfigState['allContentTypes']`